cpp/oneapi/dal/algo/basic_statistics/compute_types.cpp

```diff
@@ -0,0 +1,213 @@
+
+template <typename Task>
+const table& compute_result<Task>::get_min() const {
+    if (!get_result_options().test(result_options::min)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->min;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_max() const {
+    if (!get_result_options().test(result_options::max)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->max;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_sum() const {
+    if (!get_result_options().test(result_options::sum)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->sum;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_sum_squares() const {
+    if (!get_result_options().test(result_options::sum_squares)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->sum2;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_sum_squares_centered() const {
+    if (!get_result_options().test(result_options::sum_squares_centered)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->sum2cent;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_mean() const {
+    if (!get_result_options().test(result_options::mean)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->mean;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_second_order_raw_moment() const {
+    if (!get_result_options().test(result_options::second_order_raw_moment)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->sorm;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_variance() const {
+    if (!get_result_options().test(result_options::variance)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->varc;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_standard_deviation() const {
+    if (!get_result_options().test(result_options::standard_deviation)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->stdev;
+}
+
+template <typename Task>
+const table& compute_result<Task>::get_variation() const {
+    if (!get_result_options().test(result_options::variation)) {
+        throw domain_error(msg::this_result_is_not_enabled_via_result_options());
+    }
+    return impl_->vart;
+}
+
+template <typename Task>
+const result_option_id& compute_result<Task>::get_result_options() const {
+    return impl_->options;
+}
+
+template <typename Task>
+void compute_result<Task>::set_min_impl(const table& value) {
+    impl_->min = value;
+}
```

Comment:
Setters should also check if valid result options are set (the same behavior as getter). You can check it in k-NN in master

Reply:
thanks, fixed